### PR TITLE
IRIS-6472 | add popup with icon

### PR DESCRIPTION
**Pull request by xkxd**
https://wikia-inc.atlassian.net/browse/IRIS-6472

@Wikia/iris

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,35 +1,23 @@
...
 import './App.scss';
 
-export default class App extends Component {
-    constructor(props) {
-        super(props);
+const NOTCH_COMPENSATION = 12;
 
-        this.state = {
-            clicks: 0,
+export default class Hyperlinking extends Component {
```
Comment (latata) [inline src/components/App.js]: Please rename this file to match with class name

File: src/components/IconPopup.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { h } from 'preact';
+import WdsAvatarIconUser from 'design-system/dist/svg/wds-icons-link.svg';
+
+import './IconPopup.scss';
+
+const IconPopup = ({ position }) => (
+    <div className="pe-icon-popup" style={{ left: position.left, top: position.top }}>
```
Comment (latata) [inline src/components/IconPopup.js]: I would call popup class `pe-popup` so we could reuse it for all (both?) popups.

File: src/components/IconPopup.scss [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	z-index: 2;
+}
+
+.pe-icon-popup::after {
+	border: 8px solid transparent;
+	border-bottom-color: #fff;
+	margin-left: -8px;
+}
+
+.pe-icon-popup::before {
```
Comment (latata) [inline src/components/IconPopup.scss]: no need to have `::before`. It's for notch border which we don't need.

File: src/index.js [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
 
-import App from './components/App';
+import HyperlinkingModule from './modules/Hyperlinking';
 
 import 'quill/dist/quill.core.css';
 
 export default function create(container, options) {
-    return new Quill(container, options);
-}
+    Quill.register('modules/hyperlinking', HyperlinkingModule);
```
Comment (latata) [inline src/index.js]: I think in this function we should just call something like `render(<Quill container={container} options={options} />)` and Quill initialization should happen in preact component.

File: src/modules/Hyperlinking.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+export default class Hyperlinking {
+    constructor(quill) {
+        this.quill = quill;
+
+        this.quill.on('selection-change', this.onSelection.bind(this));
+    }
+
+    onSelection(range) {
+        if (range) {
```
Comment (latata) [inline src/modules/Hyperlinking.js]: We should render null also when range is `null`. Currently when someone click outside editor popup is not disappearing.

**Review by latata — CHANGES_REQUESTED** [review]

File: src/components/IconPopup.scss (hunk 3 above)
Comment (xkxd) [reply src/components/IconPopup.scss]: I used notch border is needed to mimic box-shadow

File: src/index.js (hunk 4 above)
Comment (xkxd) [reply src/index.js]: I'll try!

**Review by latata — APPROVED** [review]

**Pull request merged**